pick.lic

```diff
@@ -708,48 +714,101 @@ class Pick
       echo 'Bug: Tried to loot locked box...'
       return
     end
 
-    if (@settings.fill_pouch_with_box || @settings.loot_specials.empty?)
+    # pouch full, fill from lootpouch then try again
+    if Flags['full-lootpouch']
+      Flags.reset('full-lootpouch')
+
+      case DRC.bput("fill my #{@settings.gem_pouch_adjective} #{@settings.gem_pouch_noun} with my #{@settings.autoloot_container}", /You open/, /is too full to fit another gem/, /is too full to fit any more gems/, /You'd better tie it up before putting/, /You'll need to tie it up before/)        
+      when /is too full to fit another gem/, /is too full to fit any more gems/
+        swap_out_full_gempouch
+        # fill pouch from container
+        DRC.bput("fill my #{@settings.gem_pouch_adjective} #{@settings.gem_pouch_noun} with my #{@settings.autoloot_container}", /You open/, /You fill your/)
+      end
+
+      # fill pouch from box
+      case DRC.bput("fill my #{@settings.gem_pouch_adjective} #{@settings.gem_pouch_noun} with my #{box_noun}", 'You fill your', 'You open your', 'What were you referring to', /any gems/, /too full to fit/)      
+      when /you'd better tie it up/, /you'll need to tie it up before/
+        # tie pouch once there is some gems in it
+        drc.bput("tie my #{@settings.gem_pouch_adjective} #{@settings.gem_pouch_noun}", 'you tie', "it's empty?", 'has already been tied off') if @tie_gem_pouches
+        drc.bput("fill my #{@settings.gem_pouch_adjective} #{@settings.gem_pouch_noun} with my #{@settings.autoloot_container}", /you open/, /is too full to fit another gem/, /is too full to fit any more gems/, /you'd better tie it up before putting/, /you'll need to tie it up before/)    		
+      end
+    end
+
+    if (@settings.fill_pouch_with_box || @settings.loot_specials.empty?) && !(@settings.fill_pouch_with_box == false) 
       # Don't handle full pouches here, cause if we do, we can't tie them (when they're empty)
-      DRC.bput("fill my #{@settings.gem_pouch_adjective} #{@settings.gem_pouch_noun} with my #{box_noun}",
-        'You fill your', 'You open your', 'What were you referring to', /any gems/, /too full to fit/)
+      DRC.bput("fill my #{@settings.gem_pouch_adjective} #{@settings.gem_pouch_noun} with my #{box_noun}", 'You fill your', 'You open your', 'What were you referring to', /any gems/, /too full to fit/)
     end
 
-    box_items = DRCI.get_item_list(box_noun, 'look');
-    loot = box_items.map { |item| DRC.get_noun(item) }
-    echo "box items: #{box_items} - loot: #{loot}" if @debug
+    box_items = DRCI.get_item_list(box_noun, 'look');	
+    echo("boxitems = #{box_items}") if @debug
+
+    if box_items == 'There is nothing in there\.'
+      echo 'All loot gathered, dumping box'
+      return
+    end
 
+    # loot each item if there is anything left.
+    loot = box_items.map { |item| item }
+    echo "box items: #{box_items} - loot: #{loot}" if @debug
```

Comment:
🧹 This line isn't necessary, it's simply creating a new copy of `box_items` and storing the list in a second variable named `loot`, which looks like it'll collide with line 762 which expects the word `loot` to refer to the method, not a list of nouns. Also, the debug log of box items has already been done earlier at line 749
```suggestion
```